scipy/signal/_signaltools.py

```diff
@@ -3916,13 +4012,20 @@ def detrend(data: np.ndarray, axis: int = -1,
     """
     if type not in ['linear', 'l', 'constant', 'c']:
         raise ValueError("Trend type must be 'linear' or 'constant'.")
+
+    # XXX will be able to simplify after data-apis/array-api-compat#147 lands
+    if isinstance(bp, int):
+       xp = array_namespace(data)
+    else:
+       xp = array_namespace(data, bp)
```

Comment:
it has landed, if you want to simplify now?

Reply:
When a released version of array-api-compat has that fix, yes. I've amended the comment.